core/src/main/scala/org/apache/spark/TaskEndReason.scala

```diff
@@ -204,6 +213,7 @@ case object TaskResultLost extends TaskFailedReason {
 @DeveloperApi
 case object TaskKilled extends TaskFailedReason {
   override def toErrorString: String = "TaskKilled (killed intentionally)"
+  override def countTowardsTaskFailures: Boolean = false
```

Comment:
Ah sorry I was just confused by the inheritance here. Now that you mention it, it does seem like it would make sense for this to be a val though.